policies.go

```diff
@@ -1090,6 +1111,19 @@ func (d *rackAwareRR) Reset()                              {}
 func (d *rackAwareRR) KeyspaceChanged(KeyspaceUpdateEvent) {}
 func (d *rackAwareRR) SetPartitioner(p string)             {}
 
+func (d *rackAwareRR) IsDatacenterValid(datacenters []string) bool {
+	found := false
+
+	for _, dc := range datacenters {
+		if dc == d.localDC {
+			found = true
+			break
+		}
+	}
+
+	return found
```

Comment:
After https://github.com/scylladb/gocql/pull/202, 
You need to consider `disableDCFailover` here 
```suggestion
	if !d.disableDCFailover {
		return true
	}
	for _, dc := range datacenters {
		if dc == d.localDC {
			return true
		}
	}
	return false
```

Reply:
Why does it have anything to do with disableDCFailover? The whole point is not to err on the DC name. If we do or do not fail, it should NOT be due to a misconfiguration (IMHO).

Reply:
if policy allows failover to other DCs, it means that any DC is ok for it.

Reply:
Then this entire PR isn't needed...

Reply:
Are down hosts not present in system.peers?

Reply:
> if policy allows failover to other DCs, it means that any DC is ok for it. Here is the case: You have cluster with TWO DCs, and set of clients one set targets one DC , other targets other DC. Both sets of clients are allowed to failover to other DC in case of outage.
> 
> Now, outage happens on DC1 and clients that were targeting DC1 start targeting DC2 since DC1 is not available. If you are not consider disableDCFailover here, even if `disableDCFailover` is false, clients that were targeting DC1 are going to fail to create Session because this validation tell them that there is no hosts from DC1

Did the outage just wipe out the DC from the topology? I doubt that. I hope not.

Reply:
> Are down hosts not present in system.peers?

Good point, I did not track it down exactly, but my assumption is that `refreshRing` is not called at this point and all the hosts you get there are from `ClusterConfig.Hosts`

Reply:
I have changed the code to get hosts info from `s.hostSource.GetHosts()` so it should have latest information (there are some failures in tests although, I am working on fixing it right now)

Reply:
@sylwiaszunejko , thanks, but you forgot to filter them out by `cfg.filterHost(h)`

Reply:
@sylwiaszunejko , thanks, now my inital comment makes no sense and could be ignored.